code/modules/effects/sparks/spawner.dm

```diff
@@ -0,0 +1,57 @@
+// -- Spark Datum --
+/datum/effect_system/sparks
+	var/amount = 1 				// How many sparks should we make
+	var/list/spread = list()	// Which directions we should create sparks.
+	var/turf/location 			// Where the effect is
+
+// Using the spark procs is preferred to directly instancing this.
+/datum/effect_system/sparks/New(var/atom/movable/loc, var/start_immediately = TRUE, var/amt = 1, var/list/spread_dirs = list())
+	if(!loc || loc.gcDestroyed)
+		return
+
+	if (istype(loc, /turf))
+		location = loc
+	else
+		holder = loc
+
+	if (amt)
+		amount = amt
+
+	if (spread_dirs.len)
+		spread = spread_dirs
+		
+	..(start_immediately)
+
+/datum/effect_system/sparks/queue()
+	if (holder)
+		location = get_turf(holder)
+
+	return ..()
+
+/datum/effect_system/sparks/process()
+	if (!effect_master)
+		CRASH("Effect created without a master!")
```

Comment:
There's no way for proces() to be called without an effect_master. The crash check would be better placed in /datum/effect_system/New()